### PR TITLE
convert ert shuttles to use new shuttle subsystem and ert refactoring

### PR DESCRIPTION
**Pull request by spookydonut**
spawn ert shuttles on demand

### CODE REVIEW

File: code/controllers/stonedmc/subsystem/shuttle.dm [hunk 1]
```diff
@@ -392,6 +393,9 @@ SUBSYSTEM_DEF(shuttle)
 	. = FALSE
 	// load shuttle template, centred at shuttle import landmark,
 	var/turf/landmark_turf = get_turf(locate(/obj/effect/landmark/shuttle_import) in GLOB.landmarks_list)
+	if(!landmark_turf)
+		to_chat(world, "no shuttle import landmark")
```
Comment (Rohesie) [inline code/controllers/stonedmc/subsystem/shuttle.dm]: Change this debug message not to be to the world at large?

File: code/datums/emergency_calls/emergency_call.dm [hunk 2]
```diff
@@ -146,76 +146,92 @@
...
+	if(length(picked_candidates))
+		max_medics = max(round(length(members) * 0.25), 1)
+		for(var/datum/mind/M in picked_candidates)
+			members += M
+			create_member(M)
+	else
+		message_admins("ERROR: No picked candidates, aborting.")
+		return
+
+	if(auto_shuttle_launch)
```
Comment (Rohesie) [inline code/datums/emergency_calls/emergency_call.dm]:
```suggestion
	if(auto_shuttle_launch && !shuttle.auto_launch())
```
And remove the following line and fix indentation?

File: code/datums/emergency_calls/emergency_call.dm [hunk 3]
```diff
@@ -235,25 +251,13 @@
...
-
-	var/turf/spawn_loc	= pick(spawn_list)
-	if(!istype(spawn_loc))
-		return FALSE
-
-	return spawn_loc
-
+		L = pick(shuttle?.mob_spawns)
+	if(L)
+		return get_turf(L)
```
Comment (Rohesie) [inline code/datums/emergency_calls/emergency_call.dm]:
```suggestion
		return L ? get_turf(L) : FALSE
```
And remove the `if` check?

**Comment by DominikPanic** [comment]
Didn't really seem to work in-game when tested - no runtimes, ERT wouldn't spawn (it would do the no candidates found admin message).

File: code/controllers/stonedmc/subsystem/shuttle.dm (hunk 1 above)
Comment (spookydonut) [reply code/controllers/stonedmc/subsystem/shuttle.dm]: if theres no shuttle import landmark everything will break anyway

File: code/datums/emergency_calls/emergency_call.dm (hunk 3 above)
Comment (spookydonut) [reply code/datums/emergency_calls/emergency_call.dm]: less readable

File: code/datums/emergency_calls/emergency_call.dm (hunk 2 above)
Comment (spookydonut) [reply code/datums/emergency_calls/emergency_call.dm]: readability

File: code/controllers/stonedmc/subsystem/shuttle.dm (hunk 1 above)
Comment (Rohesie) [reply code/controllers/stonedmc/subsystem/shuttle.dm]: That is fine, but that's what `CRASH` is for.

File: code/datums/emergency_calls/emergency_call.dm (hunk 2 above)
Comment (Rohesie) [reply code/datums/emergency_calls/emergency_call.dm]: This doesn't really increase readability. On the contrary.

File: code/datums/emergency_calls/emergency_call.dm (hunk 3 above)
Comment (Rohesie) [reply code/datums/emergency_calls/emergency_call.dm]: Do an early return then. Make it clear what's being returned.

File: code/controllers/stonedmc/subsystem/shuttle.dm (hunk 1 above)
Comment (spookydonut) [reply code/controllers/stonedmc/subsystem/shuttle.dm]: no i mean EVERYTHING will break

File: code/datums/emergency_calls/emergency_call.dm (hunk 2 above)
Comment (spookydonut) [reply code/datums/emergency_calls/emergency_call.dm]: Disagree

File: code/datums/emergency_calls/emergency_call.dm (hunk 3 above)
Comment (spookydonut) [reply code/datums/emergency_calls/emergency_call.dm]: disagree

**Comment by Rohesie** [comment]
Extended game mode, called the ERT through the admin verb:
dreamseeker_2019-05-21_17-43-31
dreamseeker_2019-05-21_17-43-19
Seemed to fail for some reason.

File: code/datums/emergency_calls/emergency_call.dm (hunk 3 above)
Comment (spookydonut) [reply code/datums/emergency_calls/emergency_call.dm]: it'll return null which is fine

Comment (Rohesie) [reply code/datums/emergency_calls/emergency_call.dm]: Could you make that more explicit?

**Comment by Rohesie** [comment]
Same issue as last time, the second call broke.

**Pull request merged**